src/REL/d/t/d_t_sound_area.cpp

```diff
@@ -142,26 +140,31 @@ struct UnkStruct {
 
 extern "C" void fn_80337EA0(UnkStruct *);
 extern "C" void fn_80337EF0(UnkStruct *, mVec3_c &, mVec3_c &, f32);
-extern "C" int fn_8032BFB0(UnkStruct *, EGG::Vector3f &, f32 *, f32 &, int);
+extern "C" int cM3d_Len3dSqPntAndSegLine(UnkStruct *, Vec &, Vec *, f32 *, f32 *);
 
-// ???
+// Capsule
 bool dTgSndAr_c::checkAlg3(const mVec3_c &pos) {
-    f32 q[4];
-    UnkStruct unk;
+    Vec q;
 
     f32 radius = scale.x * 100.0f;
     radius = radius * radius;
-    EGG::Vector3f a = pos;
+    Vec a = pos;
+
+    UnkStruct unk;
     fn_80337EA0(&unk);
 
+    // Line between b and c
     mVec3_c b = *mRail.getPntPosForIndex(0);
     mVec3_c c = *mRail.getPntPosForIndex(1);
 
     fn_80337EF0(&unk, b, c, scale.x * 100.0f);
     f32 d;
-    if (fn_8032BFB0(&unk, a, q, d, 0)) {
+    if (cM3d_Len3dSqPntAndSegLine(&unk, a, &q, &d, nullptr)) {
```

Comment:
This is evidence the old cM3d code still uses vector types that aren't mVec3_c - this creates copies of the vector with implicit copy constructors (using GPRs rather than FPRs).

Reply:
What I find more interesting is the address of it. its already closer to the Collision code and is disconnected from the m code. I think it is safe to it was a minimal effort port xD